internal/test/registry/registry.go

```diff
@@ -1,4 +1,4 @@
-package registry // import "github.com/docker/docker/integration-cli/registry"
+package registry // import "github.com/docker/docker/internal/test/registry"
```

Comment:
Can we leave these 3 files under `integration-cli` ?

I'm not sure we want this for `integration/`. 

We should just  be running a v2 registry in a docker image instead of building a binary, or using a mock for integration testing. A lot of this seems to be an implementation of a client library, that should already exist as a library somewhere.